xilem/src/view/split.rs

```diff
@@ -0,0 +1,274 @@
+                .draggable(self.draggable)
+                .solid_bar(self.solid_bar),
+        );
+
+        (widget_pod, (child1_state, child2_state))
+    }
+
+    fn rebuild(
+        &self,
+        prev: &Self,
+        view_state: &mut Self::ViewState,
+        ctx: &mut ViewCtx,
+        mut element: xilem_core::Mut<'_, Self::Element>,
+    ) {
+        if prev.split_axis != self.split_axis {
+            widgets::Split::set_split_axis(&mut element, self.split_axis);
+        }
+
+        if prev.split_point != self.split_point {
+            widgets::Split::set_split_point(&mut element, self.split_point);
+        }
+
+        if prev.min_size != self.min_size {
+            widgets::Split::set_min_size(&mut element, self.min_size.0, self.min_size.1);
+        }
+
+        if prev.bar_size != self.bar_size {
+            widgets::Split::set_bar_size(&mut element, self.bar_size);
+        }
+
+        if prev.min_bar_area != self.min_bar_area {
+            widgets::Split::set_min_bar_area(&mut element, self.min_bar_area);
+        }
+
+        if prev.draggable != self.draggable {
+            widgets::Split::set_draggable(&mut element, self.draggable);
+        }
+
+        if prev.solid_bar != self.solid_bar {
+            widgets::Split::set_bar_solid(&mut element, self.solid_bar);
+        }
+
+        ctx.with_id(CHILD1_VIEW_ID, |ctx| {
+            let child1_element = widgets::Split::child1_mut(&mut element);
+            self.child1
+                .rebuild(&prev.child1, &mut view_state.0, ctx, child1_element);
+        });
+
+        ctx.with_id(CHILD2_VIEW_ID, |ctx| {
+            let child2_element = widgets::Split::child2_mut(&mut element);
+            self.child2
+                .rebuild(&prev.child2, &mut view_state.1, ctx, child2_element);
+        });
+    }
+
+    fn teardown(
+        &self,
+        view_state: &mut Self::ViewState,
+        ctx: &mut ViewCtx,
+        mut element: xilem_core::Mut<'_, Self::Element>,
+    ) {
+        let child1_element = widgets::Split::child1_mut(&mut element);
+        self.child1.teardown(&mut view_state.0, ctx, child1_element);
+
+        let child2_element = widgets::Split::child2_mut(&mut element);
+        self.child2.teardown(&mut view_state.1, ctx, child2_element);
+    }
+
+    fn message(
+        &self,
+        view_state: &mut Self::ViewState,
+        id_path: &[xilem_core::ViewId],
+        message: DynMessage,
+        app_state: &mut State,
+    ) -> xilem_core::MessageResult<Action, DynMessage> {
+        match id_path.split_first() {
+            Some((&CHILD1_VIEW_ID, rest)) => {
+                self.child1
+                    .message(&mut view_state.0, rest, message, app_state)
+            }
+            Some((&CHILD2_VIEW_ID, rest)) => {
+                self.child2
+                    .message(&mut view_state.1, rest, message, app_state)
+            }
+            _ => unreachable!("invalid id_path for Split view message: {:?}", id_path),
```

Comment:
We normally warn/error in these cases, e.g. https://github.com/linebender/xilem/blob/aaa84034efbdf7621ab8197f74e32ac7fb5c4bbb/xilem/src/view/spinner.rs#L89-L91, rather than panic.

We don't think it's possible to reach this state, if all other views are written correctly. But falling over isn't something we should do lightly.

Reply:
That makes sense, I've changed it.

Do you think we should do the same for the asserts when building the view?
Maybe change all asserts to `debug_assert!` and then print an error and bound the values within their valid range to handle it more gracefully in release mode?